pkg/credentialprovider/keyring.go

```diff
@@ -57,11 +57,11 @@ type lazyDockerKeyring struct {
 func (dk *BasicDockerKeyring) Add(cfg DockerConfig) {
 	if dk.index == nil {
 		dk.index = make([]string, 0)
-		dk.creds = make(map[string][]docker.AuthConfiguration)
+		dk.creds = make(map[string][]dockertypes.AuthConfig)
 	}
 	for loc, ident := range cfg {
 
-		creds := docker.AuthConfiguration{
+		creds := dockertypes.AuthConfig{
```

Comment:
> Docker uses the AuthConfig to hold the config file information (see code), so at least we know that it is feasible to replace our own type with AuthConfig. :)

Hence my question to @lavalamp.  I lean towards owning my own data format and converting so that a dependency rebase will never bork my data serialization without me knowing it, but he owns the API-ish things for kube.